src/Juvix/Prelude/Path.hs

```diff
@@ -1,28 +1,91 @@
 module Juvix.Prelude.Path
   ( module Juvix.Prelude.Path,
     module Path,
+    module Path.IO,
   )
 where
 
+import Data.List.NonEmpty qualified as NonEmpty
 import Juvix.Prelude.Base
-import Path hiding ((</>))
+import Juvix.Prelude.Path.OrphanInstances ()
+import Path hiding ((<.>), (</>))
 import Path qualified
+import Path.IO hiding (listDirRel, walkDirRel)
+
+absDir :: FilePath -> Path Abs Dir
+absDir r = fromMaybe (error ("not an absolute file path: " <> pack r)) (parseAbsDir r)
 
--- | Synonym for Path.</>. Useful to avoid name clashes
 infixr 5 <//>
 
+-- | Synonym for Path.</>. Useful to avoid name clashes
 (<//>) :: Path b Dir -> Path Rel t -> Path b t
 (<//>) = (Path.</>)
 
+someFile :: FilePath -> SomeBase File
+someFile r = fromMaybe (error ("not a file path: " <> pack r)) (parseSomeFile r)
+
+someDir :: FilePath -> SomeBase Dir
+someDir r = fromMaybe (error ("not a dir path: " <> pack r)) (parseSomeDir r)
+
 relFile :: FilePath -> Path Rel File
-relFile = fromJust . parseRelFile
+relFile r = fromMaybe (error ("not a relative file path: " <> pack r)) (parseRelFile r)
 
 relDir :: FilePath -> Path Rel Dir
-relDir = fromJust . parseRelDir
+relDir r = fromMaybe (error ("not a relative directory path: " <> pack r)) (parseRelDir r)
+
+absFile :: FilePath -> Path Abs File
+absFile r = fromMaybe (error ("not an absolute file path: " <> pack r)) (parseAbsFile r)
+
+destructAbsDir :: Path Abs Dir -> (Path Abs Dir, [Path Rel Dir])
+destructAbsDir d = go d []
+  where
+    go :: Path Abs Dir -> [Path Rel Dir] -> (Path Abs Dir, [Path Rel Dir])
+    go p acc
+      | isRoot p = (p, acc)
+      | otherwise = go (parent p) (dirname p : acc)
+
+isRoot :: Path a Dir -> Bool
+isRoot p = parent p == p
+
+-- | is the root of absolute files always "/" ?
+destructAbsFile :: Path Abs File -> (Path Abs Dir, [Path Rel Dir], Path Rel File)
+destructAbsFile x = (root, dirs, filename x)
+  where
+    (root, dirs) = destructAbsDir (parent x)
+
+isJuvixFile :: Path b File -> Bool
+isJuvixFile = (== Just ".juvix") . fileExtension
+
+isHiddenDirectory :: Path b Dir -> Bool
+isHiddenDirectory p = case toFilePath (dirname p) of
+  "./" -> False
```

Comment:
minor: I know we don't care about windows at the moment but consider using https://hackage.haskell.org/package/path-0.9.2/docs/Path-Internal-Posix.html#v:relRootFP instead of "./" literal. [source](https://github.com/commercialhaskell/path/blob/becc65ff039c55f2251480e6230213543b0d2634/src/Path/Internal/Include.hs#L66)

Reply:
done in 2a249828259abd40a5a85a10f57d5da460ea5582